lib/bundler/fetcher.rb

```diff
@@ -105,14 +88,8 @@ def specs(gem_names, source)
       end
       @use_api = false if fetchers.none?(&:api_fetcher?)
 
-      specs[remote_uri].each do |name, version, platform, dependencies|
-        next if name == 'bundler'
-        spec = nil
-        if dependencies
-          spec = EndpointSpecification.new(name, version, platform, dependencies)
-        else
-          spec = RemoteSpecification.new(name, version, platform, self)
-        end
+      specs.each do |spec|
+        next if spec.name == 'bundler'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.